src/encord_active/app/views/model_quality.py

```diff
@@ -2,87 +2,258 @@
+                    labels,
+                    predictions,
+                    model_predictions_matched,
+                )
+
+                # --- THE FOLLOWINGS WILL BE MOVED INTO page.build() METHOD LATER ---
+
+                y_true, y_pred = (
+                    list(labels_filtered[reader.ClassificationLabelSchema.class_id]),
+                    list(predictions_filtered[reader.ClassificationPredictionSchema.class_id]),
+                )
+
+                class_names = sorted(list(set(y_true).union(y_pred)))
+
+                precision, recall, f1, support = get_precision_recall_f1(y_true, y_pred)
+                accuracy = get_accuracy(y_true, y_pred)
+
+                # PERFORMANCE METRICS SUMMARY
+
+                col_acc, col_prec, col_rec, col_f1 = st.columns(4)
+                col_acc.metric("Accuracy", f"{float(accuracy):.2f}")
+                col_prec.metric(
+                    "Mean Precision",
+                    f"{float(precision.mean()):.2f}",
+                    help="Average of precision scores of all classes",
+                )
+                col_rec.metric(
+                    "Mean Recall", f"{float(recall.mean()):.2f}", help="Average of recall scores of all classes"
+                )
+                col_f1.metric("Mean F1", f"{float(f1.mean()):.2f}", help="Average of F1 scores of all classes")
+
+                # METRIC IMPORTANCE
+                if model_predictions.shape[0] > 60_000:  # Computation are heavy so allow computing for only a subset.
+                    num_samples = st.slider(
+                        "Number of samples",
+                        min_value=1,
+                        max_value=len(model_predictions),
+                        step=max(1, (len(model_predictions) - 1) // 100),
+                        value=max((len(model_predictions) - 1) // 2, 1),
+                        help="To avoid too heavy computations, we subsample the data at random to the selected size, "
+                        "computing importance values.",
+                    )
+                    if num_samples < 100:
+                        st.warning(
+                            "Number of samples is too low to compute reliable index importances. "
+                            "We recommend using at least 100 samples.",
+                        )
+                else:
+                    num_samples = model_predictions.shape[0]
+
+                metric_columns = list(get_state().predictions.metric_datas_classification.predictions.keys())
+                metric_importance_chart = create_metric_importance_charts(
+                    model_predictions_matched_filtered,
+                    metric_columns=metric_columns,
+                    num_samples=num_samples,
+                    prediction_type=MainPredictionType.CLASSIFICATION,
+                )
+                st.altair_chart(metric_importance_chart, use_container_width=True)
```

Comment:
This needs to be wrapped in a try catch as it will raise an error if there are not enough examples.